### ISSUE TITLE
Incomplete Term Assignment Choices

### ISSUE THREAD

**Issue by fantasai** [w3c/process#694]
When we have multiple-length terms in an election due to vacated seats, we assign the longer ones in preference to the candidates with more support in the election, and the shorter ones to the candidates with less support. Overall, I think this makes sense, but I'm wondering if the candidates with more support should be allowed to choose the shorter terms if they actually prefer to have them?

**Comment by dwsinger** [comment]
Interesting, but might this not lead to the outcome of the election being uncertain until after the end of some sort of haggling period between the elected candidates?

**Comment by dwsinger** [comment]
I like chaals' point about resignation.

Perhaps the simplest is simply to tell the candidates that if they want to haggle and swap in between the election results being announced, and being sworn in (seated at their first meeting), they are at liberty to do so?

**Comment by frivoal** [comment]
I am mildly concerned that this could be an opportunity for pressuring some candidates into making room for some others by having them declare they don't want the long term even if they actually did. Note that the pressure doesn't necessarily need to come from the other candidates themselves.

As a candidate who in the past, has been successfully convinced/pressured not to run in W3C a election with arguments along the lines of "X is running, and obviously, they're more of the AB caliber than you are, so what are you even doing" from people I respected, I am sensitive to this. In particular, I'll note, based on that personal experience, that it doesn't necessarily take a lot of pressure to accomplish that. Someone who's new at this, on the fence about whether they should be running at all can be swayed by apparently reasonable senior people without excessive force.

Imagine: “X has been on the AB before, and will be critical to it. They're in charge of project FOO and BAR, and have a lot of experience, a good connection with the Board / Team / Partners / WHATWG / TAG… Besides, STV rankings don't necessarily mean you're support base is that deep, it may be a narrow one. It's great that you won, but you're new to this, and you're not even sure you're going to like it in the AB, or cope well with the workload. How about you leave that stronger candidate the long term, you take the short term and see how it goes for now? You can always run again and win a full term if things pan out.”

Will it work every time? No. Might it sway someone who's already on the fence, and may feel some measure of impostor syndrome, or feels unsure they're demographic is actually welcome? I say it could.

Besides, I have occasionally heard of people who refrained from filling a formal objection because they're from a comparatively small company, and their main client would be on the receiving end of that formal objection, and hurt feelings could be too dangerous for business. What if you're a small company/IE, win the long term, and your primary contact at your main/sole client gets the short one? Given the option, could there be some unhealthy temptation (with or without pressure) to "be nice" and make room for them?

Now, maybe I'm paranoid, and none of that will happen. But if it does, it mostly won't be observable. I think I'd rather not open that door.

**Comment by cwilso** [comment]
This situation has historically been fairly rare.  I'd rather not complexify things, and simply nail down how the assignment is supposed to work.

**Comment by dwsinger** [comment]
I am *much* more concerned about the way in which we use the STV system to allocate the terms. The assumption that the election-round is an indicator of the more strongly-preferred candidates is somewhat questionable. My gut is wondering whether "total votes received" (after transfers-in but before transfers-out of excess) would be better, but I haven't analyzed it nor do I know if it's workable.

**Comment by cwilso** [comment]
The problem here is I think you would have to specify some things that are not implied by "STV" - though STV may be currently implemented that way in the implementation we use, I have no idea.

Notably, as I understand it, STV doesn't need to process the candidates in each round in any particular order; Bob might pass the quota before Julie in the same round not because he got the more votes (first-preference in the first round, but total effective votes for subsequent rounds), but because his name came up first.  (For STV's normal purposes, they would both pass the quota anyway in that round, so why would it matter?). STV isn't designed to ORDER the candidates precisely.

In short, I think in order to really ensure they are ordered, the best you could do is look inside each round, and ensure that candidates that will pass the quota that round are ordered by effective number of votes.  It's going to be messy for situations like this one.

**Comment by frivoal** [comment]
I am not sure what makes STV unordered. Isn't this appropriate:
> order by round, and within a round, order by number of votes, and if the number of votes is equal, use https://www.w3.org/2021/Process-20211102/#random to break ties

**Comment by cwilso** [comment]
You’re missing my point.  That may well BE appropriate, but it’s not “STV”
per se - it’s a hack into a possible implementation of STV, and would need
to be specified; you can’t just say “STV order.”

**Comment by frivoal** [comment]
On the one hand, sure. On the other hand, approval voting doesn't produce an other either, it merely answers the binary question of "is this person elected". But if you do want it to produce an order, the answer is pretty obvious, more votes mean a higher rank, and ties are broken as using https://www.w3.org/2021/Process-20211102/#random.

My claim is that while Meek STV also tells you who's elected and who's not, and isn't inherently an ordering system, there is also a fairly obvious way to get an order out of it.

**Comment by dwsinger** [comment]
No, I am far from convinced that round-priority is the right ordering. Happily, it's a nit and I can drop it.

**Comment by cwilso** [comment]
@frivoal yes, we would absolutely have this same problem with PROPORTIONAL approval voting, but not just Approval Voting - there is a very simple vote count in that case.

I'm contesting your statement "the answer is pretty obvious" because Meek STV does NOT enforce ordering of processing candidates inside a round where more than one candidate is elected - it doesn't have to, because it doesn't affect the results, and the NUMBER of votes is irrelevant; it's only compared to the quota, not each other.  You COULD use those vote numbers; but I have no idea if it's even easy to pull them out of the implementation we use; the goal of STV is not to order candidates by preference, merely to answer the question "who passes the bar".  I'm not opposed to this approach (though I think David might be), but it IS bespoke hacking into STV to provide some ordering, and should be clearly specified how it's done.

**Comment by dwsinger** [comment]
I suspect we are trying to achieve "give the longest terms to the most popular candidates", and I seriously doubt that that correlates well to the round in which the person passed quota, that's all. And within a round, as Chris says, what do you do then anyway? But let's drop it. Short terms are rare and we have other matters to fry.

**Comment by cwilso** [comment]
I agree this is rare.  This came up, though, when I called attention to the fact that we had two conflicting descriptions of the current AB special election resolutions in https://lists.w3.org/Archives/Member/w3c-ac-forum/2022OctDec/0203.html.  I'm not sure we've answered the question of "how do we intend to resolve this".

To quote from there:

I noted a conflict in the texts of this announcement and the vote itself,
that may be important.  (I'll remind everyone that there are 3 6-month
terms and one 18-month term up in this special election.) This notice says:

   - When ballots are processed, the candidate who first reaches the
   threshold to be elected will be assigned the term ending in June 2024; the
   remaining three elected candidates will be assigned the terms ending in
   June 2023.

The ballot itself says (in section 3, "Resolving ties and short terms"):

   - After all elected individuals have been identified, when N people are
   eligible for M (less than N) incomplete terms. In this case, only the names
   of those N individuals are provided as input to the procedure. The
   incomplete terms are assigned in result order.

The latter says (I think) the 18-month term will be randomly assigned to
one of the successful candidates.  The former says "the first who reaches
the threshold" will be assigned the 18-month term.  (This is a little
confusing, as there is no explicit single "first" winner in STV - IIUC,
there may be multiple winners in each "round", and there's no explicit
ordering of those winners.  I would suggest it should be made explicit that
order comes from the number of votes to a candidate *in that round*, with
random resolution if more than one candidate has the same top number of
votes.)

**Comment by dwsinger** [comment]
Sure, I agree, be explicit, that the sorting is 
a) by round
b) if two or more people get elected in the same round, by vote count
c) if two or more people get the same vote count, by random selection

whether that is right we can leave to another day...

**Comment by frivoal** [comment]
I think we should reject the question as initially posed (can candidates chose a shorter term despite a better ranking), for the reasons in https://github.com/w3c/process/issues/694#issuecomment-1374299692.

That does leave us with the question of how to determine a ranking. I suspect the suggestion in https://github.com/w3c/process/issues/694#issuecomment-1377610194 is likely what we want to do.
